nixos/modules/services/misc/memos.nix

```
@@ -0,0 +1,192 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  cfg = config.services.memos;
+in
+{
+  options.services.memos = {
+    enable = lib.mkEnableOption "Memos note-taking";
+    package = lib.mkPackageOption pkgs "memos" { };
+
+    openFirewall = lib.mkEnableOption "opening the ports in the firewall";
```

IIUC, it is common for firewall ports to be configured by default, at least where doing so is likely to be what most users want. However, `mkEnableOption` sets `default = false`.

**If you want** it on-by-default, you could either:


```suggestion
    openFirewall = lib.mkEnableOption "opening the ports in the firewall" // {
      default = true;
      example = false;
    };
```

Or:

```suggestion
    openFirewall = lib.mkOption {
      type = lib.types.bool;
      description = "Whether to enable opening the ports in the firewall.";
      default = true;
      example = false;
    };
```

---

I'm not a fan of, by default, opening the ports of the firewall for the service : this could lead to an uncontrolled opened port on a server if you aren't careful. As this is a webapp, most of the time it will be behind a reverse proxy to be handled by something else, thus avoiding the need to open the port to the internet, as `127.0.0.1` is always permitted in the firewall. I'm of course opened to discussion about this !

---

> As this is a webapp, most of the time it will be behind a reverse proxy to be handled by something else, thus avoiding the need to open the port to the internet, as `127.0.0.1` is always permitted in the firewall.

This answers my caveat: _"at least where doing so is likely to be what most users want"_

Sounds like off-by-default is the right choice.